Framework/DataHandling/src/LoadSESANS.cpp

```diff
@@ -0,0 +1,419 @@
+*/
+ColumnMap LoadSESANS::consumeData(std::ifstream &infile, std::string &line,
+                                  int &lineNum) {
+  std::string numberRegex = "(-?\\d+(\\.\\d+)?(E-?\\d+)?)";
+
+  std::getline(infile, line);
+  std::vector<std::string> columnHeaders = split(line);
+
+  // Make sure all 4 mandatory columns have been supplied
+  for (std::string header : mandatoryColumnHeaders)
+    if (std::find(columnHeaders.begin(), columnHeaders.end(), header) ==
+        columnHeaders.end())
+      throwFormatError(line, "Failed to supply mandatory column header: \"" +
+                                 header + "\"",
+                       lineNum);
+
+  // static_cast is safe as realistically our file is never going to have enough
+  // columns to overflow
+  std::string rawRegex = "^\\s*" +
+                         repeatAndJoin(numberRegex, "\\s+",
+                                       static_cast<int>(columnHeaders.size())) +
+                         "\\s*$";
+  boost::regex lineRegex(rawRegex);
+
+  // Tokens in a line
+  std::vector<std::string> tokens;
+
+  // Map of column name -> column values
+  ColumnMap columns;
+
+  while (std::getline(infile, line)) {
+    lineNum++;
+
+    if (boost::regex_match(line, lineRegex)) {
+      tokens = split(line);
+
+      for (unsigned int i = 0; i < tokens.size(); i++)
+        columns[columnHeaders[i]].push_back(std::stod(tokens[i]));
+    } else {
+      g_log.warning("Line " + std::to_string(lineNum) +
+                    " discarded, as it was badly formed. Expected " +
+                    std::to_string(columnHeaders.size()) +
+                    " numbers, but got \"" + line + "\"");
+    }
+  }
+  g_log.information("Loaded " +
+                    std::to_string(columns[columnHeaders[0]].size()) +
+                    " rows of data");
+  return columns;
+}
+
+/**
+* Split a header into a key-value pair delimited by whitespace, where the first
+* token is the key and the remainder the value
+* @param line A string containing the line to split
+* @param lineNum The number of the line in the file, used for error messaging
+* @return A key-value pair
+* @throw runtime_error If the line contains less than two tokens
+*/
+std::pair<std::string, std::string>
+LoadSESANS::splitHeader(const std::string &line, const int &lineNum) {
+  std::pair<std::string, std::string> attribute;
+  auto i = line.begin();
+
+  // Discard leading whitespace
+  i = find_if(i, line.end(), &notSpace);
+  // Find the end of the first word
+  auto j = find_if(i, line.end(), &space);
+
+  if (j == line.end())
+    throwFormatError(line, "Expected key-value pair", lineNum);
+
+  attribute.first = std::string(i, j);
+
+  // Find start of second word
+  i = find_if(j, line.end(), notSpace);
+  if (i == line.end())
+    throwFormatError(line, "Expected key-value pair", lineNum);
+
+  // Grab from there to the end of the line
+  attribute.second = std::string(i, line.end());
+
+  return attribute;
+}
```

Comment:
`boost::trim` neatens things up nicely, but I'd prefer to avoid splitting, as for attributes like `DataFileTitle` where there are many words in a line, we're then just making more work for ourselves by having to join again